grype/pkg/package.go

```diff
@@ -101,21 +103,22 @@ func (p Package) String() string {
 	return fmt.Sprintf("Pkg(type=%s, name=%s, version=%s, upstreams=%d)", p.Type, p.Name, p.Version, len(p.Upstreams))
 }
 
-func removePackagesByOverlap(catalog *pkg.Collection, relationships []artifact.Relationship) *pkg.Collection {
+func removePackagesByOverlap(sbm *sbom.SBOM) *pkg.Collection {
+	catalog := sbm.Artifacts.Packages
 	byOverlap := map[artifact.ID]artifact.Relationship{}
-	for _, r := range relationships {
+	for _, r := range sbm.Relationships {
 		if r.Type == artifact.OwnershipByFileOverlapRelationship {
 			byOverlap[r.To.ID()] = r
 		}
 	}
 
 	out := pkg.NewCollection()
-
+	comprehensiveDistroFeed := distroFeedIsComprehensive(sbm.Artifacts.LinuxDistribution)
 	for p := range catalog.Enumerate() {
 		r, ok := byOverlap[p.ID()]
 		if ok {
 			from, ok := r.From.(pkg.Package)
-			if ok && excludePackage(p, from) {
+			if ok && excludePackage(comprehensiveDistroFeed, p, from) {
```

Comment:
nit: does it make sense to short circuit earlier on the `comprehensiveDistroFeed`? Why do we carry this check all the way into where we inspect the package?